### ISSUE TITLE
how to set username and password?

### ISSUE THREAD

**Issue by songsh** [kevinzg/facebook-scraper#456]
i see the readme, no mention the username, passwor, how to set?

**Comment by neon-ninja** [comment]
Use the parameter credentials, and pass a tuple. For example, `get_posts("Nintendo", credentials=("user@mail.com","password"))`

**Comment by jorgegene** [comment]
Is it still working? I've been using the credentials of a new create account and it just pops with unsuccesful login error.

**Comment by neon-ninja** [comment]
Still working fine for me

**Comment by AndreaWozzup** [comment]
question: if I set the credentials parameter, can I avoid adding the cookies?

i.e.
facebook_scraper.get_page_info(identifier, credentials=(user, password))

I get LoginRequired exception:
facebook_scraper.exceptions.LoginRequired: A login (cookies) is required to see this page

**Comment by neon-ninja** [comment]
`get_page_info` doesn't support the credentials parameter the same way as `get_posts`, you would probably want to decouple it, like so:

```python
from pprint import pprint
from facebook_scraper import _scraper, get_page_info
_scraper.login(user, password)
pprint(get_page_info(identifier))
```

**Comment by AndreaWozzup** [comment]
Thanks, so if I provide user and password I can avoid passing cookies?

**Comment by neon-ninja** [comment]
Yes

**Comment by AndreaWozzup** [comment]
I have a couple more questions, and probably very python-noob (sorry!), still related to login with user and password.
Let's say I want to use multiple accounts for scraping, and the scraping may happen concurrently for the different accounts.

So I initialize a dictionary of scrapers, something like this:

    users = ["user1@something.com", "user2@something.com", "user3@something.com"]

    scrapers = {}

    for user in users:
        scraper = FacebookScraper()
        for i in range(3):
            try:
                scraper.login(user, <user password>)
                if scraper.is_logged_in():
                    print("User " + user + " logged in.")
                    scrapers[user] = scraper
                    break
            except LoginError:
                pass
            except TemporarilyBanned:
                print("User " + user + " is temporarily banned.")
                break


1) Do I get to have multiple instances of scraper class in scrapers dict that stay logged in and I can use concurrently, and thread safe?

2) Moreover, usually the first login attempt always fails, it succeeds only on second attempt, is this normal?

**Comment by neon-ninja** [comment]
1. I think so, it looks fine to me. Should work, but keep an eye on the interpreted datetimes, as I don't think dateparser is thread safe
2. First login works fine for me

**Comment by null15** [comment]
For some reason I get login unsuccessful?

`from facebook_scraper import get_posts

for post in get_posts('113060560534111', pages=5, credentials=("******@hotmail.com","*****")):
    print(post['text'][:50])
`

  File "C:\Users\Raiko\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\facebook_scraper\facebook_scraper.py", line 567, in login
    raise exceptions.LoginError("Login unsuccessful")
facebook_scraper.exceptions.LoginError: Login unsuccessful

**Comment by neon-ninja** [comment]
Can you login with those credentials in an incognito window on m.facebook.com?

**Comment by null15** [comment]
> Can you login with those credentials in an incognito window on m.facebook.com?

How could I attempt this?

**Comment by neon-ninja** [comment]
Go to https://m.facebook.com/ in your browser, in an incognito session if you're using Chrome or inprivate session if you're using Firefox

**Comment by neon-ninja** [comment]
Is a login_error.html file generated in your current working directory?

**Comment by null15** [comment]
> Can you login with those credentials in an incognito window on m.facebook.com?

Yeah, I can login with my credentials on incognito

**Comment by null15** [comment]
> Is a login_error.html file generated in your current working directory? If so, can you upload it?

https://file.io/RkYpBoHffUQu

**Comment by neon-ninja** [comment]
It says that file has been deleted. Can you just upload it to this thread?

**Comment by null15** [comment]
> It says that file has been deleted. Can you just upload it to this thread?
[login_error.zip](https://github.com/kevinzg/facebook-scraper/files/7275175/login_error.zip)

**Comment by null15** [comment]
> It says that file has been deleted. Can you just upload it to this thread directly, by drag & drop?

It wasn't supported so I had to zip it

**Comment by neon-ninja** [comment]
The error message is in Swedish, which means your login page was in Swedish too. The login function is currently only designed to work with the English version of the m.facebook.com login page.

**Comment by null15** [comment]
> The error message is in Swedish, which means your login page was in Swedish too. The login function is currently only designed to work with the English version of the m.facebook.com login page.

How can I make it not be Swedish?

**Comment by neon-ninja** [comment]
Use a proxy or VPN from an English speaking country?

**Comment by null15** [comment]
> Use a proxy or VPN from an English speaking country?

Changed to U.S.A now, the error I get now is "login approval needed"

**Comment by neon-ninja** [comment]
Now you need to approve the new login from a web browser that is logged into that account. You should have got a notification

**Comment by null15** [comment]
> Now you need to approve the new login from a web browser that is logged into that account. You should have got a notification

I approved it 2 times, but when I run the command prompt again, it asks for login approval again

**Comment by null15** [comment]
Maybe I will be forced to use the cookies file? Using NordVPN to USA just keeps asking for login approval, even though I approved myself many times now

**Comment by neon-ninja** [comment]
Maybe try this? https://github.com/kevinzg/facebook-scraper/commit/840c5252e4c22bda2f4506596a244b2325fefc92

**Comment by null15** [comment]
> Maybe try this? [7aa3fb1](https://github.com/kevinzg/facebook-scraper/commit/7aa3fb10df3c59c7c334239779595dbb66d1bc17)

What should I do? try again?

**Comment by neon-ninja** [comment]
First install the latest master branch so that you get that commit with `pip install git+https://github.com/kevinzg/facebook-scraper.git`. Then try again.

**Comment by null15** [comment]
> First install the latest master branch so that you get that commit with `pip install git+https://github.com/kevinzg/facebook-scraper.git`. Then try again.

So tried 4 times, had to approve on all 4
First time: had to change PW
other 3 times: logged in after confirming with the number sent to my mail

**Comment by neon-ninja** [comment]
Sounds like it might be easier to just use cookies then

**Comment by null15** [comment]
> Sounds like it might be easier to just use cookies then. Probably FB detects that thousands of people use that IP address so don't mark it as trusted even when you approve the login.

Yeah possible, I changed to UK now instead, now I get login unsuccessful, so I suppose only U.S would work

**Comment by null15** [comment]
update: chose canada and it works!
